lib/lightning/runs/prom_ex_plugin.ex

```diff
@@ -275,4 +283,73 @@ defmodule Lightning.Runs.PromExPlugin do
 
     query |> Repo.aggregate(:count)
   end
+
+  defp project_metrics(unclaimed_threshold_seconds, period_in_seconds) do
+    Polling.build(
+      :lightning_run_project_metrics,
+      period_in_seconds * 1000,
+      {__MODULE__, :project_metrics, [unclaimed_threshold_seconds]},
+      [
+        last_value(
+          @impeded_project_event ++ [:count],
+          description:
+            "The count of projects impeded due to lack of worker capacity"
+        )
+      ],
+      detach_on_error: false
+    )
+  end
+
+  def project_metrics(unclaimed_threshold_seconds) do
+    if repo = Process.whereis(Repo) do
+      check_repo_state(repo)
+
+      threshold_time =
+        DateTime.add(Lightning.current_time(), -unclaimed_threshold_seconds)
+
+      count =
+        projects_with_available_runs_older_than(threshold_time)
+        |> projects_with_spare_capacity()
+        |> Enum.count()
+
+      :telemetry.execute(
+        @impeded_project_event,
+        %{count: count},
+        %{}
+      )
+    end
+  end
+
+  def projects_with_available_runs_older_than(threshold_time) do
+    query =
+      from r in Run,
+        join: w in assoc(r, :workflow),
+        join: p in assoc(w, :project),
+        where: r.state == :available,
+        where: r.inserted_at <= ^threshold_time,
```

Comment:
`inserted_at` on runs is not indexed but an `explain` run on prod shows that the query builder prefers to use the index on `state`, so I am not going to add an index on `inserted_at` for now.